shared/Microsoft.Azure.Devices.Shared/Twin.cs

```diff
@@ -52,7 +55,66 @@ public Twin(string deviceId) : this()
         /// <summary>
         /// Twin's Version
         /// </summary>
-        public long Version { get; set; }
+        [DefaultValue(null)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public long? Version { get; set; }
```

Comment:
These are public API additions that will affect Shared and so DeviceClient, ServiceClient and ProvisioningServiceClient.

@tameraw  @jasmineymlo @mamokarz  FYI